src/features/device-state/routes/state.ts

```diff
@@ -329,6 +330,38 @@ const stateQuery = _.once(() =>
 	}),
 );
 
+/**
+ * We can allow device keys to query state as root because it's a controlled query
+ * and we know that if they can access this device they can access its state
+ */
+const canQueryStateAsRoot = async ({
+	uuid,
+	req,
+	tx,
+}: {
+	uuid: string;
+	req: permissions.PermissionReq;
+	tx: Tx;
+}) => {
+	if (typeof req.apiKey?.key !== 'string') {
+		return false;
+	}
+	// Must be a device key
+	if (!(await isApiKeyWithRole(req.apiKey?.key, 'device-api-key'))) {
+		return false;
+	}
+	// And must have access to the device
+	const device = api.resin.prepare<{ uuid: string }>({
```

Comment:
Shouldn't this be

```suggestion
	const device = await api.resin.prepare<{ uuid: string }>({
```

?